### PR TITLE
fix: Adding equals and hashcode for CronSchedule

### PR DESCRIPTION
**Pull request by 0x01F4**
## Brief, plain english overview of your changes here

Adding Equals and HasCode for CronSchedule

## Reminders
- [x] Added/ran automated tests
- [x] Update README and/or examples
- [x] Ran `mvn spotless:apply`

---
cc @kagkarlsson

### CODE REVIEW

**Comment by 0x01F4** [comment]
@kagkarlsson Can you please merge this?

File: db-scheduler/src/test/java/com/github/kagkarlsson/scheduler/task/CronTest.java [hunk 1]
```diff
@@ -125,7 +127,23 @@ public void should_mark_schedule_as_disabled() {
...
+    assertNotEquals(
+        new CronSchedule("1 * * * *", ZoneId.systemDefault(), CronStyle.UNIX),
+        new CronSchedule("* * * * *", ZoneId.systemDefault(), CronStyle.UNIX));
+    assertNotEquals(
+        new CronSchedule("1 * * * *", london, CronStyle.UNIX),
+        new CronSchedule("* * * * *", ZoneId.systemDefault(), CronStyle.UNIX));
+    assertNotEquals(
+        new CronSchedule("* * * * * *", ZoneId.systemDefault(), CronStyle.SPRING53),
+        new CronSchedule("* * * * * *", ZoneId.systemDefault(), CronStyle.SPRING));
   }
```
Comment (kagkarlsson) [inline db-scheduler/src/test/java/com/github/kagkarlsson/scheduler/task/CronTest.java:147]: Could we also add an EqualsVerifier test?

```java
  @Test
  public void equals_and_hash_code() {
    EqualsVerifier.forClass(Daily.class).verify();
  }
  ```

**Comment by kagkarlsson** [comment]
Thank you for contributing. Minor comment

File: db-scheduler/src/test/java/com/github/kagkarlsson/scheduler/task/CronTest.java (hunk 1 above)
Comment (0x01F4) [reply db-scheduler/src/test/java/com/github/kagkarlsson/scheduler/task/CronTest.java]: Sure added.

File: db-scheduler/src/test/java/com/github/kagkarlsson/scheduler/task/CronTest.java [hunk 2]
```diff
@@ -125,7 +128,28 @@ public void should_mark_schedule_as_disabled() {
...
+  public void validate_cron_schedule_equals() {
+    assertEquals(
+        new CronSchedule("* * * * *", ZoneId.systemDefault(), CronStyle.UNIX),
+        new CronSchedule("* * * * *", ZoneId.systemDefault(), CronStyle.UNIX));
+    assertNotEquals(
+        new CronSchedule("1 * * * *", ZoneId.systemDefault(), CronStyle.UNIX),
+        new CronSchedule("* * * * *", ZoneId.systemDefault(), CronStyle.UNIX));
+    assertNotEquals(
+        new CronSchedule("1 * * * *", london, CronStyle.UNIX),
+        new CronSchedule("* * * * *", ZoneId.systemDefault(), CronStyle.UNIX));
```
Comment (kagkarlsson) [inline db-scheduler/src/test/java/com/github/kagkarlsson/scheduler/task/CronTest.java:144]: This one was prob not supposed to have variation in cron-pattern. (and maybe not use zone-default here, test will fail in London, no?)

File: db-scheduler/src/test/java/com/github/kagkarlsson/scheduler/task/CronTest.java [hunk 3]
```diff
@@ -125,7 +128,28 @@ public void should_mark_schedule_as_disabled() {
...
+    assertNotEquals(
+        new CronSchedule("* * * * * *", ZoneId.systemDefault(), CronStyle.SPRING53),
+        new CronSchedule("* * * * * *", ZoneId.systemDefault(), CronStyle.SPRING));
+  }
+
+  @Test
+  public void equals_and_hash_code() {
+    EqualsVerifier.forClass(CronSchedule.class).verify();
```
Comment (kagkarlsson) [inline db-scheduler/src/test/java/com/github/kagkarlsson/scheduler/task/CronTest.java:152]: Nice 👍

File: db-scheduler/src/test/java/com/github/kagkarlsson/scheduler/task/CronTest.java (hunk 2 above)
Comment (0x01F4) [reply db-scheduler/src/test/java/com/github/kagkarlsson/scheduler/task/CronTest.java]: Yes my bad fixed it.

**Review by kagkarlsson — APPROVED** [review]
Not sure what the use-case is here, but can't hurt to have equals and hashcode, thanks! LGTM

**Comment by 0x01F4** [comment]
> Not sure what the use-case is here, but can't hurt to have equals and hashcode, thanks! LGTM

I want reschedule on config change but when we reschedule next execution time change.So I want to check if schedule is same then we don't reschedule.

**Comment by 0x01F4** [comment]
@kagkarlsson can you merge this?

**Comment by kagkarlsson** [comment]
🎉 This issue has been resolved in `v15.1.0` ([Release Notes](https://github.com/kagkarlsson/db-scheduler/releases/tag/v15.1.0))

**Pull request merged**